deployments/traefik.go

```diff
@@ -166,24 +166,21 @@ func (k Traefik) Deploy(ctx context.Context, c *kubernetes.Cluster, ui *termui.U
 	//  c  | no      | yes       | Something has claimed the namespace | Error
 	//  d  | no      | no        | Namespace is free for use           | Deploy
 
-	// Bug: The current order of checking (namespace first, then
-	// service), and associated actions, mishandles case A (Error
-	// instead of nothing).
+	log.Info("check presence, local service")
 
-	log.Info("check presence, namespace")
-
-	_, err = c.Kubectl.CoreV1().Namespaces().Get(
+	_, err = c.Kubectl.CoreV1().Services(TraefikDeploymentID).Get(
 		ctx,
-		TraefikDeploymentID,
+		"traefik",
 		metav1.GetOptions{},
 	)
 	if err == nil {
```

Comment:
This is assuming that any error means the Service is missing. I think we should also check that if there is an error, it is about the service not being there, otherwise exit with an error. Something like:

```
if err == nil {
...
} else if !apierrors.IsNotFound(err)` {
  return err
}
```

Reply:
same thing below

Reply:
Sure. Can do that.

Reply:
Done. Commit 416156b7b1867e1d0570dcffc5c4b2dae3d3e1a7